include/libopencm3/nrf/51/ficr.h

```diff
@@ -0,0 +1,73 @@
+/** @defgroup ficr_defines FICR Defines
+ *
+ * @brief <b>Defined Constants and Types for the NRF51xx Factory information
+ * configuration registers</b>
+ *
+ * @ingroup NRF51xx_defines
+ *
+ * @version 1.0.0
+ *
+ * @date Nov 2021
+ *
+ *LGPL License Terms @ref lgpl_license
+ */
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2017-2018 Unicore MX project<dev(at)lists(dot)unicore-mx(dot)org>
+ * Copyright (C) 2021 Eduard Drusa <ventyl86(at)netkosice(dot)sk>
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <libopencm3/cm3/common.h>
+#include <libopencm3/nrf/memorymap.h>
+#include <libopencm3/nrf/common/ficr.h>
+
+/* Deprecated */
+#define FICR_CLENR0         MMIO32(FICR_BASE + 0x028)
+/* Deprecated */
+#define FICR_PPFC           MMIO32(FICR_BASE + 0x02C)
+
+#define FICR_NUMRAMBLOCK    MMIO32(FICR_BASE + 0x034)
+#define FICR_SIZERAMBLOCKS  MMIO32(FICR_BASE + 0x038)
+
+/* Deprecated */
+#define FICR_SIZERAMBLOCK(n)    MMIO32(FICR_BASE + 0x038 + 0x4 * (n))
```

Comment:
let's just delete it then right? we're not bringing in deprecated things.

Reply:
This should read Deprecated by Nordic on nRF52. Yet still perfectly usable on both nRF51 and nRF52.